packages/eas-cli/src/build/metadata.ts

```diff
@@ -27,15 +30,13 @@ export async function collectMetadataAsync<T extends Platform>(
     workflow: ctx.workflow,
     credentialsSource: ctx.buildProfile.credentialsSource,
     sdkVersion: ctx.exp.sdkVersion,
-    runtimeVersion:
-      (await resolveRuntimeVersionAsync({
-        exp: ctx.exp,
-        platform: ctx.platform,
-        workflow: ctx.workflow,
-        projectDir: ctx.projectDir,
-        env: ctx.buildProfile.env,
-        cwd: ctx.projectDir,
-      })) ?? undefined,
+    runtimeVersion: runtimeMetadata?.runtimeVersion,
+    fingerprintSource: runtimeMetadata?.fingerprintGCSBucketKey
+      ? {
+          type: FingerprintSourceType.GCS,
+          bucketKey: runtimeMetadata?.fingerprintGCSBucketKey,
+        }
+      : undefined,
```

Comment:
I believe we should also try to add support for `fingerprintSource.type === PATH` in this PR. 

If you want to check if a build is run using local mode you can check if `buildCtx.localBuildOptions.localBuildMode` is defined.